oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/writer/DefaultIndexWriterFactory.java

```diff
@@ -50,14 +64,30 @@ public LuceneIndexWriter newInstance(IndexDefinition def, NodeBuilder definition
                 "Expected %s but found %s for index definition",
                 LuceneIndexDefinition.class, def.getClass());
 
-        LuceneIndexDefinition definition = (LuceneIndexDefinition)def;
+        LuceneIndexDefinition definition = (LuceneIndexDefinition) def;
 
-        if (mountInfoProvider.hasNonDefaultMounts()){
-            return new MultiplexingIndexWriter(directoryFactory, mountInfoProvider, definition,
-                    definitionBuilder, reindex, writerConfig);
+        if (mountInfoProvider.hasNonDefaultMounts()) {
+            return wrapWithPipelinedIndexWriter(
+                    new MultiplexingIndexWriter(directoryFactory, mountInfoProvider, definition, definitionBuilder, reindex, writerConfig),
+                    definition.getIndexName());
         }
-        return new DefaultIndexWriter(definition, definitionBuilder, directoryFactory,
+        DefaultIndexWriter writer = new DefaultIndexWriter(definition, definitionBuilder, directoryFactory,
                 FulltextIndexConstants.INDEX_DATA_CHILD_NAME,
                 LuceneIndexConstants.SUGGEST_DATA_CHILD_NAME, reindex, writerConfig);
+
+        return wrapWithPipelinedIndexWriter(writer, definition.getIndexName());
+    }
+
+    public void close() {
+        LOG.info("Closing LuceneIndexWriterFactory");
+        if (indexWriterPool == null) {
+            LOG.info("Not using an Index writer pool");
+        } else {
+            indexWriterPool.close();
+        }
```

Comment:
I am not sure if incremental indexing opens/closes a writer every time it kicks in.
In that case, we will see a lot of these logs. Should we lower the debug level for the log messages here?